src/main/scala/com/wavesplatform/account/KeyPair.scala

```diff
@@ -0,0 +1,36 @@
+package com.wavesplatform.account
+
+import com.google.common.collect.Interners
+import com.wavesplatform.common.state.ByteStr
+import com.wavesplatform.common.utils.Base58
+import com.wavesplatform.crypto
+import com.wavesplatform.transaction.ValidationError.GenericError
+
+import scala.util.{Failure, Success}
+
+final case class KeyPair(seed: ByteStr) {
+  private[this] lazy val pair = crypto.createKeyPair(seed)
+
+  lazy val privateKey: PrivateKey = PrivateKey(pair._1)
+  lazy val publicKey: PublicKey   = PublicKey(pair._2)
+}
+
+object KeyPair {
+  private[this] val interner = Interners.newWeakInterner[KeyPair]()
```

Comment:
Unlike public keys, key pairs are not instantiated so often, so we probably don't need an interner here.